### PR TITLE
Memory error in DSPKernel

### PR DESCRIPTION
**Pull request by ryanfrancesconi**
There was a double freed error happening in DSPKernel.hpp 
(*** error for object 0xbaddc0dedeadbead: pointer being freed was not allocated) 

This was because the destroy() method of AKAmplitudeTrackerDSPKernel was being explicitly called but the destructor was also being triggered, the same pointer was being freed twice in both the destructor and in the destroy(). Please check this as I'm seeing some weird behavior about when the destroy() methods are being called. I'm seeing destroy() being called on objects that I never instantiated personally! 

This fix solves the crash I was seeing, but perhaps there are other situations? 

This crash is related to adding deinit { engine.detach() } to AKNode as well as the previous memory fix I did for the self in closures for AUv3. Reason being, I believe beforehand because the memory was being leaked and the objects held onto - these destroy methods were never called. This is my theory anyway! Once things are properly released, this bug was revealed. BOOM!

### CODE REVIEW

**Comment by aure** [comment]
@ryanfrancesconi This is passing the test and it looks fine but I just wanted to also ask @adamnemecek to look at since he introduce the AKSporthKernel.  Also, you've changed AKAmplitudeTracker but wouldn't a lot of the nodes be affected in exactly the same way?  I've got a couple of other comments which I'll add to the lines directly.

File: AudioKit/Common/Nodes/Playback/Player/AKAudioPlayer.swift [hunk 1]
```diff
@@ -236,6 +236,7 @@ open class AKAudioPlayer: AKNode, AKToggleable {
     }
     
     deinit {
+        AKLog("* AKAudioPlayer")
```
Comment (aure) [inline AudioKit/Common/Nodes/Playback/Player/AKAudioPlayer.swift]: Is this descriptive enough?  I for one wouldn't know what this meant when it appeared in the console.

File: AudioKit/Common/Nodes/AKNode.swift [hunk 2]
```diff
@@ -46,6 +46,7 @@ extension AVAudioConnectionPoint {
     }
     
     deinit {
+        //AKLog("* AKNode")
```
Comment (aure) [inline AudioKit/Common/Nodes/AKNode.swift]: I'm see a lot of these added commented out prints/logs.  Can't they just be comments?  Were they just around for debug?  Not a big deal, just thought I'd bring it up.  (also, lets always use AKLog and not print so it can be suppressed more easily).

File: AudioKit/Common/Nodes/Analysis/Amplitude Tracker/AKAmplitudeTracker.swift [hunk 3]
```diff
@@ -82,6 +82,10 @@ open class AKAmplitudeTracker: AKNode, AKToggleable, AKComponent {
         halfPowerPointParameter?.setValue(Float(halfPowerPoint), originator: token!)
     }
 
+    deinit {
+        AKLog("* AKAmplitudeTracker")
```
Comment (aure) [inline AudioKit/Common/Nodes/Analysis/Amplitude Tracker/AKAmplitudeTracker.swift]: Again, is this descriptive enough?  / Do we need it only for debug?

File: AudioKit/Common/Nodes/Playback/Player/AKAudioPlayer.swift (hunk 1 above)
Comment (ryanfrancesconi) [inline AudioKit/Common/Nodes/Playback/Player/AKAudioPlayer.swift]: Interestingly when AKLog actually outputs that it shows up as:
deinit * AKAudioPlayer

That's pretty clear.

File: AudioKit/Common/Nodes/AKNode.swift (hunk 2 above)
Comment (ryanfrancesconi) [inline AudioKit/Common/Nodes/AKNode.swift]: Because AKNode is in a bunch of spots, I didn't leave that in - however in debugging the AmplitudeTracker deinit it was helpful to see when it was called. Technically, we don't need it in there! Could just remove it.

File: AudioKit/Common/Nodes/Analysis/Amplitude Tracker/AKAmplitudeTracker.swift (hunk 3 above)
Comment (ryanfrancesconi) [inline AudioKit/Common/Nodes/Analysis/Amplitude Tracker/AKAmplitudeTracker.swift]: same deal, shows up as:
deinit * AKAmplitudeTracker

**Comment by ryanfrancesconi** [comment]
The memory release is in the DSPKernel class, so all the subclasses would inherit this change. I didn't change AKAmplitudeTracker in this pull request in any code way other than comments. The previous changes I made I had to go into each AUv3 class and make changes.

**Comment by ryanfrancesconi** [comment]
One other comment that I forgot to mention in the pull request. I am seeing the ~destructor being triggered everytime, but I don't see the destroy() method called everytime. I only see destroy() called if I connect up a bunch of units. If not as many connections were made, it just does the ~. I didn't trace this too deep as it seemed enough for me to simply deallocate &sp in the destructor only.

**Comment by ryanfrancesconi** [comment]
But yes, please whoever wrote that stuff, please check it as maybe I'm not understanding the sources of the crashes properly.

File: AudioKit/Common/Nodes/Analysis/Amplitude Tracker/AKAmplitudeTracker.swift (hunk 3 above)
Comment (ryanfrancesconi) [inline AudioKit/Common/Nodes/Analysis/Amplitude Tracker/AKAmplitudeTracker.swift]: yes, only need for debugging

**Comment by aure** [comment]
Thanks Ryan!

**Comment by ryanfrancesconi** [comment]
You got it! Hope I am understanding everything - but it seems good to me!
